src/affordance/tasks/sequential_interpreter.py

```diff
@@ -1,688 +1,213 @@
-            if result:
-                return result, code_input
-        except:
-            pass
-
-    try:
-        # Run code snippet as is, if it was generated by code_generate
-        # More robust outputs
-        result = subprocess.run(
-            [sys.executable, "-c", code_input],
-            capture_output=True,
-            text=True,
-            timeout=60,
-        )
-        if result.stderr == "":
-            simplified_result = simplify_ans(result.stdout)
-            return simplified_result, code_snippet
-    except:
-        pass
-
-    try:
-        # Run arithmetic python code generate
-        code_output = safe_execute(code_snippet)
-        result = simplify_ans(code_output)
-        if result:
-            return result, code_snippet
-    except:
-        pass
-
-    try:
-        # Run code snippet as is, if it was generated by code_generate
-        result = subprocess.run(
-            [sys.executable, "-c", code_snippet],
-            capture_output=True,
-            text=True,
-            timeout=60,
-        )
-        if result.stderr == "":
-            simplified_result = simplify_ans(result.stdout)
-            return simplified_result, code_snippet
-    except:
-        pass
-
-    if "error" in code_snippet or (code_input and "error" in code_input):
-        return code_error(code_snippet, code_input)
-
-    try:
-        # Command instruction (if its code) starts in the next line.
-        code_snippet = code_snippet.split("\n", 1)[1]
-        result = subprocess.run(
-            [sys.executable, "-c", code_snippet],
-            capture_output=True,
-            text=True,
-            timeout=60,
-        )
-        if result.stderr == "":
-            simplified_result = simplify_ans(result.stdout)
-            return simplified_result, code_snippet
-    except:
-        return None, code_snippet
-
-    return None, code_snippet
-
-
-def code_generate_then_execute(instruction, code_input):
-    # output = openai.Edit.create(
-    # model="code-davinci-edit-001",
-    # input="x = " + code_input,
-    # instruction="Python code for " + instruction,
-    # temperature=0,
-    # top_p=1
-    # )
-    # code_snippet = output.choices[0].text
-
-    code_snippet, generate_details = code_generate(instruction=instruction, code_input=code_input)
-    time.sleep(2)
-    # result = subprocess.run([sys.executable, "-c", code_snippet], capture_output=True, text=True)
-    # if result.stderr == "":
-    #     return result.stdout, code_snippet
-    result = code_execute(code_snippet=code_snippet)
-    return result
-
-
-def code_error(code_snippet, code_input):
-    # Command instruction (if its code) starts in the next line.
-    try:
-        code_snippet = code_snippet.split("\n", 1)[1]
-        result = subprocess.run(
-            [sys.executable, "-c", code_snippet],
-            capture_output=True,
-            text=True,
-            timeout=60,
+class TopDownVisitor:
```

Comment:
what's the difference between TopDownVisitor and TopDownVisitorBeta? with so many changes, it's hard to check the diff